src/components/split_button/split_button.tsx

```diff
@@ -0,0 +1,289 @@
+import {
+  OuiSplitButtonControl,
+  OuiSplitButtonControlProps,
+  OuiSplitButtonOption,
+} from './split_button_control';
+import { OuiInputPopover } from '../popover';
+import {
+  OuiContextMenuItem,
+  OuiContextMenuItemLayoutAlignment,
+} from '../context_menu';
+import { keys } from '../../services';
+import { OuiI18n } from '../i18n';
+
+enum ShiftDirection {
+  BACK = 'back',
+  FORWARD = 'forward',
+}
+
+export type OuiSplitButtonProps = CommonProps &
+  Omit<
+    OuiSplitButtonControlProps,
+    'onChange' | 'onClick' | 'options' | 'value'
+  > & {
+    /**
+     * Pass an array of options : string or ReactNode
+     */
+    options?: OuiSplitButtonOption[];
+
+    /**
+     * optional index of options item to mark with checkmark
+     */
+    selectedIndex?: string;
+
+    /**
+     * Classes for the context menu item
+     */
+    itemClassName?: string;
+
+    /**
+     * You must pass an `onChange` function to handle selection of an option item
+     */
+    onChange?: (index: number) => void;
+
+    /**
+     * You must pass `onClick` function to handle click of the Primary button
+     */
+    onClick?: () => void;
+
+    /**
+     * Change to `true` if you want horizontal lines between options.
+     * This is best used when options are multi-line.
+     */
+    hasDividers?: boolean;
+
+    /**
+     * Change `OuiContextMenuItem` layout position of icon
+     */
+    itemLayoutAlign?: OuiContextMenuItemLayoutAlignment;
+
+    /**
+     * Applied to the outermost wrapper (popover)
+     */
+    popoverClassName?: string;
+
+    /**
+     * Controls whether the options are shown. Default: false
+     */
+    isOpen?: boolean;
+  };
+
+export const OuiSplitButton = ({
+  fullWidth,
+  options = [],
+  selectedIndex,
+  hasDividers,
+  itemClassName,
+  onChange,
+  onClick,
+  isOpen: propIsOpen,
+  className,
+  popoverClassName,
+  children,
+  ...rest
+}: OuiSplitButtonProps) => {
+  const itemNodes: Array<HTMLButtonElement | null> = [];
+  const [isOpen, setIsOpen] = useState(propIsOpen || false);
+
+  let selectedIndexInt: number | undefined = Number(selectedIndex);
+  if (Number.isNaN(selectedIndexInt)) selectedIndexInt = undefined;
+
+  const openPopover = () => {
+    setIsOpen(true);
+
+    const focusSelected = () => {
+      requestAnimationFrame(() => {
+        if (selectedIndexInt != null) {
+          focusItemAt(selectedIndexInt);
+        } else {
+          focusSelected();
+        }
+      });
+    };
+
+    requestAnimationFrame(focusSelected);
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      openPopover();
+    }
+  });
+
+  const setItemNode = (node: HTMLButtonElement | null, index: number) => {
+    itemNodes[index] = node;
+  };
+
+  const closePopover = () => {
+    setIsOpen(false);
+  };
+
+  const itemClicked = (index: number) => {
+    setIsOpen(false);
+    if (onChange) {
+      onChange(index);
+    }
+  };
+
+  const onSelectKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) => {
```

Comment:
I couldnt get this to work. just to be sure this should open the dropdown when the up or down aro is pressed when the user is focused on the chevron right?